cmd/remote/main.go

```diff
@@ -0,0 +1,365 @@
+	if err != nil {
+		return instanceDetails, err
+	}
+
+	err = json.Unmarshal(body, &instanceDetails)
+	return instanceDetails, err
+
+}
+
+func getOutput(url string) ([]byte, error) {
+	var output instanceOutput
+
+	req, err := http.NewRequest(
+		"GET",
+		url,
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if apiKey != "" {
+		req.Header.Add("apikey", apiKey)
+	} else if authToken != "" {
+		req.Header.Add("Direktiv-Token", authToken)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(body, &output)
+	if err != nil {
+		return nil, err
+	}
+
+	outputStr, err := base64.StdEncoding.DecodeString(output.Data)
+	return outputStr, err
+
+}
+
+func updateRemoteWorkflow(url string, localPath string) error {
+	localWorkflowFile, err := os.Open(localPath)
+	if err != nil {
+		return fmt.Errorf("failed to open input file: %v", err)
+	}
+
+	localWorkflowBytes, err := ioutil.ReadAll(localWorkflowFile)
+	if err != nil {
+		return fmt.Errorf("failed to read input file: %v", err)
+	}
+
+	localWorkflowData := bytes.NewBuffer(localWorkflowBytes)
+
+	req, err := http.NewRequest(
+		"POST",
+		url,
+		localWorkflowData,
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create request file: %v", err)
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	if apiKey != "" {
+		req.Header.Add("apikey", apiKey)
+	} else if authToken != "" {
+		req.Header.Add("Direktiv-Token", authToken)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to send request: %v", err)
+	}
+
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("failed to update workflow, server responsed with %s", resp.Status)
+	}
+
+	return err
+}
+
+var rootCmd = &cobra.Command{
+	Use:   "dre WORKFLOW_PATH",
+	Short: "Remotely execute direktiv workflows with local files. This process will update your latest remote workflow to your local WORKFLOW_PATH file",
+	Long: `Remotely execute direktiv workflows with local files. This process will update your latest remote workflow to your local WORKFLOW_PATH file.
+
+EXAMPLE: dre helloworld.yaml --addr http://192.168.1.1 --namespace admin --path helloworld`,
+	Args: cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		// Load Config From flags / config
+		addr = viper.GetString("addr")
+		path = viper.GetString("path")
+		namespace = viper.GetString("namespace")
+		apiKey = viper.GetString("api-key")
+		authToken = viper.GetString("auth-token")
+
+		http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: insecure}
+
+		instanceStatus := "pending"
+		urlPrefix := fmt.Sprintf("%s/api/namespaces/%s", addr, namespace)
+		urlUpdateWorkflow := fmt.Sprintf("%s/tree/%s?op=update-workflow", urlPrefix, strings.TrimPrefix(path, "/"))
+
+		cmd.Printf("Updating Namespace: '%s' Workflow: '%s'\n", namespace, path)
```

Comment:
Logs & status messages are good, but we should print them to stderr. That way stdout remains valid JSON if they don't use the flag that would otherwise send it to a file. Improves ability to script.